HAL.md

```diff
@@ -0,0 +1,117 @@
+# Hardware abstract layer(HAL) overview
+
+## Abstract 
+Hardware abstract layer(HAL) defines a standard interface for hardware vendors to implement, which enables Tangle-Accelerator to be aware of lower-level driver implementations. HAL layer support such as uart, GPIO, secure storage and other low-level operators used in TA-endpoint. The TA-endpoint is a daemon which base on Linux which offer monitor for device and can send the message to tangle-accelerator. HAL layer lets user only needs to implement their device operators from abstract interface which defined inside `device.h` but not need to know how TA-endpoint work. 
+
+## Directory Overview
+* connectivity: Connect to Tangle-accelerator functions.
+* devices: The real directory to implement devices.
+* hal: Hardware abstraction layer.
+* hooks: Git hook functions.
+* tests: Unit-tests.
+* third_party: Third party directory.
+* utils: Utility functions for TA-endpoint.
+
+## Create directory for new device
+Create directory for new device into `devices`.
+```
+mkdir devices/<my_device>
+```
+* Create `impl.c` and `Makefile` under new device directory.
+* Include `device.h` headers into new device header file.
```

Comment:
Should we mention the header file `impl.h`?

Reply:
`impl.h` is not necessary. If users need it, they can create it by themselves.

Reply:
But the `impl.h` is mentioned at the line `78` of the example code of `impl.c` in the `HAL.md`.